src/NuGet.Core/NuGet.ProjectModel/DependencyGraphSpec.cs

```diff
@@ -14,6 +14,8 @@ namespace NuGet.ProjectModel
 {
     public class DependencyGraphSpec
     {
+        public static readonly string DGSpecFileName = "{0}.nuget.dgspec.json";
```

Comment:
This is not the file name.  It's a format string.  Instead of exposing this as a `static readonly` field (or better yet `const`) and requiring callers to format it, how about moving this format string inside a static method:

    public static string GetDgSpecFileName(string projectName)
    {
        ...
    }

Then callers won't have to [know how to format it](https://github.com/NuGet/NuGet.Client/pull/2521/files#diff-66b6645a4772b600680f93c7fee07f5aR242).

Reply:
I like it, will change it accordingly